text/0000-format.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: format
+- Start Date: 2016-07-27
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the seperation of the `string()` method and formatting. This wil remove formatting from the builtin package of the standard library and seperate the disperate concerns of producing a string representation of an object and formatting strings.
+
+# Motivation
+
+String formatting should not be involved in the string representation of an object. Pony's string formatting should also have improved documentation since it differs from that of most other languages.
+
+# Detailed design
+
+- Redefine the Stringable interface:
+```pony
+interface box Stringable
+  """
+  Things that can be turned into a String.
+  """
+  fun string(): String
```

Comment:
Note that the current standard is for `string` to return `String iso`, rather than `String val`.  I'm not sure which of these I prefer at the moment - I have mixed feelings because they have their advantages and disadvantages.

Using `iso` is nice when the string is being built from scratch on every call, because it gives more flexibility in the caller - it can become any other capability when consumed.

Using `val` is nice when the string doesn't need to be built from scratch on every call, because it can avoid unnecessary copies.  For example, a class could hold a `String val` internally that is returned whenever `string` is called.  If the return value of the interface were `String iso`, then a new string would have to be allocated and copied on every call, since you can't get an `iso` reference from a `val` one, and even if you could, it couldn't be shared across multiple calls.